### PR TITLE
[CodeCompletion] Update for SE-0345 shorthand optional binding

### PR DESCRIPTION
**Pull request by rintaro**
Suggest visible optional values after 'if let'.

rdar://90399603

cc: @calda

### CODE REVIEW

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Review by calda — COMMENTED** [review]
Awesome, thanks!

File: lib/IDE/CompletionLookup.cpp [hunk 1]
```diff
@@ -3116,3 +3119,35 @@ void CompletionLookup::getStmtLabelCompletions(SourceLoc Loc, bool isContinue) {
...
+          DynamicLookupInfo dynamicLookupInfo) -> bool {
+        auto *VarD = dyn_cast<VarDecl>(VD);
+        if (!VarD)
+          return false;
+
+        auto Ty = getTypeOfMember(VD, dynamicLookupInfo);
+        if (!Ty->isOptional())
+          return false;
+
+        return true;
```
Comment (ahoppen) [inline lib/IDE/CompletionLookup.cpp:3143]: This could be simplified to 

```suggestion
        return Ty->isOptional();
```

**Review by ahoppen — APPROVED** [review]
One minor stylistic suggestion. Otherwise LGTM.

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Pull request merged**